scripts/hooks/tech-lead-authoring-guard.py

```
@@ -199,30 +199,124 @@ def _extract_mutation_targets(command: str):
+# Kwarg form: ``open('path', mode='w')`` / ``open('path', buffering=0, mode='wb')``.
+# Path captured from first positional; mode captured from the kwarg. The gap
+# between the path and the ``mode=`` token allows any non-paren chars (so
+# additional kwargs like ``buffering=`` or ``encoding=`` don't block the
+# match). Code-reviewer tightening #4: positional form alone let the kwarg
+# spelling slip through as a read-only call.
+_OPEN_WRITE_KWARG_RE = re.compile(
+    r"""open\(\s*['"]([^'"]+)['"][^)]*\bmode\s*=\s*['"]([^'"]*[waxA+][^'"]*)['"]"""
+)
+
+
+def _extract_write_targets_from_body(body: str):
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `_extract_write_targets_from_body`

(ANN202)